sdk/java-sdk/src/main/scala/kalix/javasdk/impl/ProxyInfoHolder.scala

```diff
@@ -75,10 +69,14 @@ class ProxyInfoHolder(system: ExtendedActorSystem) extends Extension {
   def identificationInfo: Option[IdentificationInfo] = _identificationInfo
 
   def proxyPort: Option[Int] = {
+    // If portOverride is filled, we choose it. Otherwise we use the announced one.
+    // Note: with old proxy versions `proxyInfo.proxyPort` will default to 0
+    val chosenPort = _portOverride.getOrElse(_proxyAnnouncedPort)
```

Comment:
Previously, we were fixing the `_chosenPort` when we were receive the `ProxyInfo`. We can't do this for the integration tests as the port is only available afterwards. Therefore, each time we call `proxyPort`, we must calculate the `chosenPort`.